docs: improve docs of elements using DataItem interface (#948)
Refinitiv/refinitiv-ui: documents/src/pages/elements/overlay-menu.md

## Data property interface

> The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) type for its data items. Each item is `ItemData` type extended from [DataItem](./custom-components/utils/data-management#data-item). 

Reviewer: ```suggestion
The `data` property of the `ef-overlay-menu` uses the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) type for its data items. Each item is `ItemData` type extended from [DataItem](./custom-components/utils/data-management#data-item). 
```

> The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) interface for its data items. Each item is ItemData type which is extended from [DataItem](./custom-components/utils/data-management#data-item). 

Reviewer 2: ```suggestion
The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) interface for its data items. Each item is `ItemData` type which is extended from [DataItem](./custom-components/utils/data-management#data-item). 
```

> The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) interface for its data items. Each item is `ItemData` type which is extended from [DataItem](./custom-components/utils/data-management#data-item). 

Reviewer: this one is not defaulted but fixed to `ItemData`.
```suggestion
The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) type for its data items. Each item is `ItemData` type extended from [DataItem](./custom-components/utils/data-management#data-item). 
```

> The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) interface that extend from Array of [ItemData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/item/helpers/types.ts) or [CollectionComposer](https://ui.refinitiv.com/custom-components/utils/data-management#collection-composer).

Reviewer 3: ```suggestion
The `data` property of the `ef-overlay-menu` use the `OverlayMenuData` interface that extend from Array of [ItemData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/item/helpers/types.ts) or [CollectionComposer](https://ui.refinitiv.com/custom-components/utils/data-management#collection-composer).
```

> The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) interface that extend from Array of [ItemData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/item/helpers/types.ts) or [CollectionComposer](https://ui.refinitiv.com/custom-components/utils/data-management#collection-composer).

Reviewer 2: ```suggestion
The `data` property of the `ef-overlay-menu` use the [OverlayMenuData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/overlay-menu/helpers/types.ts) interface that either extends from Array of [ItemData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/item/helpers/types.ts) or 
utilising [CollectionComposer](https://ui.refinitiv.com/custom-components/utils/data-management#collection-composer) to provide data.
```
